python/tvm/contrib/hexagon/pytest_plugin.py

```diff
@@ -56,14 +56,19 @@ def _compose(args, decs):
 requires_hexagon_toolchain = tvm.testing.requires_hexagon(support_required="compile-only")
 
 
-@pytest.fixture(scope="session")
 def android_serial_number() -> Optional[str]:
+    """Return the android serial number or simulator"""
     serial = os.getenv(ANDROID_SERIAL_NUMBER, default="")
     # Setting ANDROID_SERIAL_NUMBER to an empty string should be
     # equivalent to having it unset.
     if not serial.strip():
         serial = None
-    return serial
+    if serial == "simulator":
+        return serial
+    else:
+        # Split android serial numbers into a list
+        serial = serial.split(",")
```

Comment:
This will fail if serial was set to `None` at L65.

Reply:
If the serial was set to None, we skip the test with a message that says android serial number is not set.

Reply:
Line 61 gets the serial number from an environment variable, then if it's an empty string, it's set to `None` at line 65.  This `None` can get to line 70...

Reply:
So in the new pr, I have changed this that if the android serial number is set to empty string, it would skip the test with an error of "ANDROID_SERIAL_NUMBER is not set".

Reply:
Ah, you're right, I was looking at the old sources.